values.yaml

```diff
@@ -171,6 +171,11 @@ client:
   # This should be set to true if connectInject or meshGateway is enabled.
   grpc: false
 
+  # exposeGossipPorts exposes the clients' gossip ports as hostPorts.
+  # This is only necessary if pod ips in the k8s cluster are not directly
+  # routable and the Consul servers are outside of the k8s cluster.
+  exposeGossipPorts: false
```

Comment:
should this be `exposeGossipPort` (singular) since there's only one port?

Reply:
There are actually two [ports](https://www.consul.io/docs/agent/options.html#ports) used for Serf. These are `8301` and `8302` for `lan` and `wan` respectively.

Reply:
The wan port is only for the servers though. I see we have it here but it's not used by the agents. This is because servers are the only ones that will gossip outside of their datacenter (i.e. to other datacenters).

Reply:
From these [docs](https://kubernetes.io/docs/concepts/configuration/overview/#services): `each <hostIP, hostPort, protocol> combination must be unique`. Technically we're exposing just 8301, but because we need to expose both tcp and udp as hostPorts, we have to specify them separately. I figured it would be confusing to leave this singular given that there are two definitions.

Reply:
It's the same port though, just different protocols.

It's not a blocker, just when I read "exposes the clients' gossip ports" I think there's two ports being exposed but there isn't, there's just one: 8301.